lib/ansible/module_utils/urls.py

```diff
@@ -716,10 +720,15 @@ def open_url(url, data=None, headers=None, method=None, use_proxy=True,
             handlers.append(authhandler)
 
         elif username and force_basic_auth:
-            if headers is None:
-                headers = {}
+            headers["Authorization"] = basic_auth_header(username, password)
 
-            headers["Authorization"] = "Basic %s" % base64.b64encode("%s:%s" % (username, password))
+        else:
+            rc = netrc.netrc(os.environ.get('NETRC'))
+            login = rc.authenticators(parsed[1])
```

Comment:
If `NETRC` is not in the environment, then `.get` will return `None`.

`netrc`'s behaviour is to:

 > If no argument is given, the file .netrc in the user’s home directory will be read

You have however pointed out the scenario there `$HOME/.netrc` does not exist, in which case it throws `IOError: [Errno 2] No such file or directory: '/Users/jsok/.netrc'`

I'll push a fix now to ensure we don't propagate that exception!